doc: Update documentation for volume clone (#14190)
rook/rook: Documentation/Storage-Configuration/Ceph-CSI/ceph-csi-volume-clone.md

In [pvc-clone](https://github.com/rook/rook/tree/master/deploy/examples/csi/rbd/pvc-clone.yaml),
`dataSource` should be the name of the `PVC` which is already created by RBD
CSI driver. The `dataSource` kind should be the `PersistentVolumeClaim`.

> The `storageClassName` can be any storageclass (not necessarily same as Parent PVC) with the

Reviewer: !!! note

Reviewer: The `storageClassName` can be any RBD storageclass (not necessarily same as Parent PVC) with the

following conditions:

> * `provisinor` should be same for both the Parent PVC and the Clone PVC.

Reviewer: does clone doesnt work cross operator? LIke in a DR operation where there maybe two rook operator?

Author: In the case of 2 different rook operators, the provisioner name will be different too right? hence it won't work for that case

Reviewer: so we can say clone doesn't work cross-cluster....

> * The non-encrypted PVC cannot be cloned to an encrypted one and vise-versa.

Reviewer: can we have a high level chart,

encrypted -> encrypted  possible 
encrypted -> non-encrypted not possible 
and soo on....

In [pvc-clone](https://github.com/rook/rook/tree/master/deploy/examples/csi/rbd/pvc-clone.yaml),
`dataSource` should be the name of the `PVC` which is already created by RBD
CSI driver. The `dataSource` kind should be the `PersistentVolumeClaim`.
The `storageClassName` can be any RBD storageclass (not necessarily same as Parent PVC)

> !!!note

Reviewer 2: The notes are usually just short unformatted paragraphs. If using bullets, let's not use the "!!! note". How about this?
```suggestion
Please note:
```

>     * `provisinor` should be same for both the Parent PVC and the Clone PVC.

Reviewer 2: What if the provisioner is not the same? Is there a scenario for that? If not, let's use "must" instead of "should"
```suggestion
    * `provisioner` must be same for both the Parent PVC and the Clone PVC.
```

>     * `provisinor` must be the same for both the Parent PVC and the Clone PVC.

Reviewer 2: typo?
```suggestion
    * `provisioner` must be the same for both the Parent PVC and the Clone PVC.
```

Author: oops, thanks for the catch

>     * `provisioner` must be the same for both the Parent PVC and the Clone PVC.

Author: sure, I will add it as a different commit here itself

>     * `provisioner` must be the same for both the Parent PVC and the Clone PVC.

Author: @Madhu-1 https://rook.io/docs/rook/latest/Storage-Configuration/Ceph-CSI/ceph-csi-snapshot/ there is no mentioned of using same sc  for restore/snapshots. I don't think we need to add further details there.

>     * `provisioner` must be the same for both the Parent PVC and the Clone PVC.

Reviewer 3: i think it worded differently but that possible one can create PVC from encrypted storageclass and try to restore to non-encrypted storageclass.

>     * The non-encrypted PVC cannot be cloned to an encrypted one and vise-versa.

Reviewer 2: ```suggestion
    * The non-encrypted PVC cannot be cloned to an encrypted one and vice-versa.
```

In [pvc-clone](https://github.com/rook/rook/tree/master/deploy/examples/csi/cephfs/pvc-clone.yaml),
`dataSource` should be the name of the `PVC` which is already created by CephFS

> CSI driver. The `dataSource` kind should be the `PersistentVolumeClaim`.

Reviewer 3: Lets not update cephfs section as nothing is tested or support yet.

The `storageClassName` can be any CephFS storageclass (not necessarily same as Parent PVC)

> !!!note

Reviewer 2: same as above

>     * `provisinor` should be same for both the Parent PVC and the Clone PVC.

Reviewer 2: ```suggestion
    * `provisioner` must be the same for both the Parent PVC and the Clone PVC.
```

>     * The non-encrypted PVC cannot be cloned to an encrypted one and vise-versa.

Reviewer 2: ```suggestion
    * The non-encrypted PVC cannot be cloned to an encrypted one and vice-versa.
```
